swarm/network/stream/intervals_test.god

```diff
@@ -298,20 +299,20 @@ func newTestExternalClient(netStore *storage.NetStore) *testExternalClient {
 	}
 }
 
-func (c *testExternalClient) NeedData(ctx context.Context, hash []byte) func(context.Context) error {
-	wait := c.netStore.FetchFunc(ctx, storage.Address(hash))
-	if wait == nil {
-		return nil
+func (c *testExternalClient) NeedData(ctx context.Context, hash []byte) (bool, func(context.Context) error) {
+	fi, loaded, ok := c.netStore.GetOrCreateFetcherItem(ctx, hash, "syncer")
+	if !ok {
+		return loaded, nil
 	}
-	select {
-	case c.hashes <- hash:
-	case <-ctx.Done():
-		log.Warn("testExternalClient NeedData context", "err", ctx.Err())
-		return func(_ context.Context) error {
-			return ctx.Err()
+
+	return loaded, func(ctx context.Context) error {
+		select {
+		case <-fi.Delivered:
+		case <-time.After(timeouts.SyncerClientWaitTimeout):
+			return fmt.Errorf("chunk not delivered through syncing after 20sec. ref=%s", fmt.Sprintf("%x", hash))
```

Comment:
hardcoded timeout text. convert to `%d` based on `timeouts.SyncerClientWaitTimeout`

Reply:
These tests are commented out. I am not sure what they are testing, I was hoping for someone to bring this during code review (`.god` stands for `go-dontcompile`). If they are not needed, we should remove them.